### DISCUSSION TITLE
onDisconnect - Connect Modal Shows automatically, and can't be closed

### DISCUSSION THREAD

**Discussion by lgtmrobo** [rainbow-me/rainbowkit#1009] [Q&A]
When I use wagmi disconnect, the connect modal automatically pops up and is not able to be closed. Any idea why this might happen, and how to fix it? I would not like it to pop up on disconenct, and I should also be able to close it.

https://user-images.githubusercontent.com/50972198/217675949-1648fac8-7a77-4446-8b11-c8a0e71c276f.mov

**Comment by etherealiska** [comment]
This is happening to me as well, and would like to be able to disable automatically opening (and be able to close it)

**Reply by lgtmrobo** [reply]
I was able to get the fix in for being able to close it here [https://github.com/rainbow-me/rainbowkit/issues/1010](https://github.com/rainbow-me/rainbowkit/issues/1010) although it still pops up when I disconnect
